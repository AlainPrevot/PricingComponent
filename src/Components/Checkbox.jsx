import '../Sass/Checkbox.sass';
import PropTypes from 'prop-types';

export const Checkbox = ({ onMode }) => {
	return (
		<div className='container'>
			<span>Annually</span>
			<div className='checkbox'>
				<input type='checkbox' id='checkbox' onChange={() => onMode()} />
				<label htmlFor='checkbox'></label>
			</div>
			<span>Monthly</span>
		</div>
	);
};

Checkbox.propTypes = {
	onMode: PropTypes.func.isRequired,
};
