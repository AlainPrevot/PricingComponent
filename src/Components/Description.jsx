import PropTypes from 'prop-types'
import '../Sass/Description.sass'

export const Description = ({Storage, Users, Memory}) => {
  return (
    <div className='Card--Description border-top-bottom'>
        <span>{Storage} TB Storage</span>
        <span className='border-top-bottom'>{Users} Users Allowed</span>
        <span>Send up to {Memory} GB</span>
    </div>
  )
};

Description.proptypes = {
    Storage: PropTypes.number.isRequired,
    Users: PropTypes.number.isRequired,
    Memory: PropTypes.number.isRequired,
};

