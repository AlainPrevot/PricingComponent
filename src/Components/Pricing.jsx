import '../Sass/Pricing.sass';
import PropTypes from 'prop-types';

export const Pricing = ({ value }) => {
	return (
		<div className='card--pricing'>
			<span>$</span>
			<h1>{value}</h1>
		</div>
	);
};

Pricing.propTypes = {
	value: PropTypes.number.isRequired,
};
