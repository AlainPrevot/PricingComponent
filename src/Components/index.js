export * from './Card';
export * from './Checkbox';
export * from './ContainerCard';
export * from './Description';
export * from './Pricing';
export * from './Title';
export * from './TypePlan';