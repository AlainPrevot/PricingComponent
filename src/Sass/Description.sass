.card--description
    display: flex
    flex-direction: column
    list-style: none
    margin: 8px 0
    width: 100%

    li
        box-sizing: border-box
        width: 100%
        text-align: center
        padding: 16px
        font-size: 0.9rem

.border-top-bottom
    border-top: 1px solid var(--separation)
    border-bottom: 1px solid var(--separation)
