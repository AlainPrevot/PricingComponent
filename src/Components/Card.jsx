import PropTypes from 'prop-types';
import { Description } from './Description';
import { Pricing } from './Pricing';
import '../Sass/Card.sass';
import { usePlan } from '../Hooks/usePlan';

export const Card = ({
	title,
	pricing,
	storage,
	users,
	memory,
	theme,
	plane,
}) => {
	const { newPricing } = usePlan(pricing, plane);

	return (
		<div className='card' data-theme={theme}>
			<h2>{title}</h2>
			<Pricing value={newPricing} />
			<Description Storage={storage} Memory={memory} Users={users} />
			<button className='card--button'>LEARN MORE</button>
		</div>
	);
};

Card.propTypes = {
	title: PropTypes.string.isRequired,
	pricing: PropTypes.number.isRequired,
	storage: PropTypes.number.isRequired,
	memory: PropTypes.number.isRequired,
	users: PropTypes.number.isRequired,
	theme: PropTypes.string.isRequired,
	plane: PropTypes.string.isRequired,
};
