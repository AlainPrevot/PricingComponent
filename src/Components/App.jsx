import { Card } from './Card';
import '../Sass/App.sass';
import { Title } from './Title';
import { Checkbox } from './Checkbox';
import { useState } from 'react';

export const App = () => {
	const [plan, setPlan] = useState('monthly');

	const TypePlan = () => {
		setPlan(plan === 'monthly' ? 'Annually' : 'monthly');
	};

	return (
		<div
			style={{
				display: 'flex',
				alignItems: 'center',
				height: '76vh',
				// justifyContent: 'space-around',
				flexDirection: 'column',
			}}
			className='cuerpo'
		>
			<Title title={'Our Pricing'} />
			<Checkbox onMode={TypePlan} />
			<div className='prueba'>
				<Card
					title={'Basic'}
					pricing={19.99}
					storage={500}
					memory={3}
					users={2}
					theme={'secundary'}
					plane={plan}
				/>
				<Card
					title={'Professional'}
					pricing={24.99}
					storage={1}
					memory={10}
					users={5}
					theme={'primary'}
					plane={plan}
				/>
				<Card
					title={'Master'}
					pricing={39.99}
					storage={2}
					memory={20}
					users={10}
					theme={'secundary'}
					plane={plan}
				/>
			</div>
		</div>
	);
};
