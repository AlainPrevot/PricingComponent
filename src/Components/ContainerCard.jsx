import PropTypes from 'prop-types';
import { Card } from '../Components';
import '../Sass/ContainerCard.sass';
import data from '../Data/Data.js';

export const ContainerCard = ({ plan }) => {
	return (
		<div className='container--card'>
			{data.map(data => (
				<Card
					key={data.id}
					title={data.title}
					pricing={data.pricing}
					storage={data.storage}
					memory={data.memory}
					users={data.users}
					theme={data.theme}
					plane={plan}
				/>
			))}
		</div>
	);
};

ContainerCard.propTypes = {
	plan: PropTypes.string.isRequired,
};
