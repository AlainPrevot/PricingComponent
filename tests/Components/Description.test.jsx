import { render, screen } from '@testing-library/react';
import { Description } from '../../src/Components/Description';

describe('Pruebas en Description', () => {
	const storageTest = 10;
	const usersTest = 20;
	const memoryTest = 30;

	test('Debe de hacer match con el snapshot', () => {
		const { container } = render(
			<Description
				storage={storageTest}
				users={usersTest}
				memory={memoryTest}
			/>
		);
		expect(container).toMatchSnapshot();
	});

	test('Debe de mostrar las propiedades pasadas', () => {
		render(
			<Description
				storage={storageTest}
				users={usersTest}
				memory={memoryTest}
			/>
		);

		expect(screen.getByText(`${storageTest} GB Storage`)).toBeTruthy();
		expect(screen.getByText(`${usersTest} Users Allowed`)).toBeTruthy();
		expect(screen.getByText(`Send up to ${memoryTest} GB`)).toBeTruthy();
	});

	test('La unidad del storage debe ser GB al pasar un numero mayor de 2 digitos', () => {
		render(<Description storage={500} users={usersTest} memory={memoryTest} />);

		expect(screen.getByText('500 GB Storage')).toBeTruthy();
	});

	test('La unidad del storage debe ser TB al pasar un numero de 1 digito', () => {
		render(<Description storage={3} users={usersTest} memory={memoryTest} />);

		expect(screen.getByText('3 TB Storage')).toBeTruthy();
	});
});
