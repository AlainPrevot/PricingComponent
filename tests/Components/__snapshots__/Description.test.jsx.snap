// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Description Debe de hacer match con el snapshot 1`] = `
<div>
  <ul
    class="card--description border-top-bottom"
  >
    <li>
      10
       
      GB
       Storage
    </li>
    <li
      class="border-top-bottom"
    >
      20
       Users Allowed
    </li>
    <li>
      Send up to 
      30
       GB
    </li>
  </ul>
</div>
`;
